vizro-core/examples/scratch_dev/components.py

```diff
@@ -0,0 +1,96 @@
+"""Contains custom components used inside the dashboard."""
+
+from typing import Annotated, Literal
+
+import dash_bootstrap_components as dbc
+import vizro.models as vm
+from dash import Input, Output, State, clientside_callback, html
+from pydantic import BeforeValidator, conlist
+from vizro.models import VizroBaseModel
+from vizro.models._models_utils import _log_call, check_captured_callable_model
+from vizro.models.types import ComponentType
+
+
+class FlexContainer(VizroBaseModel):
+    """Container to group together a set of components on a page.
+
+    Args:
+        type (Literal["container"]): Defaults to `"container"`.
+        components (list[ComponentType]): See [ComponentType][vizro.models.types.ComponentType]. At least one component
+            has to be provided.
+        title (str): Title to be displayed.
+
+    """
+
+    type: Literal["flex_container"] = "flex_container"
+    # TODO[mypy], see: https://github.com/pydantic/pydantic/issues/156 for components field
+    components: conlist(  # type: ignore[valid-type]
+        Annotated[ComponentType, BeforeValidator(check_captured_callable_model)],
+        min_length=1,
+    )
+
+    @_log_call
+    def build(self):
+        """Returns custom flex container."""
+        return html.Div(
+            id=self.id, children=[component.build() for component in self.components], className="flex-container"
+        )
+
+
+class CollapsibleContainer(vm.Container):
+    """Container to group together a set of components on a page.
+
+    Args:
+        type (Literal["collapse_container"]): Defaults to `"collapse_container"`.
+
+    """
+
+    type: Literal["collapse_container"] = "collapse_container"
+    is_open: bool = True
+
+    @_log_call
+    def build(self):
+        """Returns custom collapsible container."""
+        clientside_callback(
+            """
+            (n_clicks, is_open) => {
```

Comment:
This callback all looks good to me 👍 Let's put it in its own js file when this goes beyond a POC, just like we do for other clientside callbacks.

I think it's correct that it doesn't share code with `collapse_nav_panel` but it would be nice to add some comments explaining their similarities/differences. This took me a while to figure out just reading through code now.